### DISCUSSION TITLE
Implementing wikidata SERVICE label syntax

### DISCUSSION THREAD

**Discussion by D063520** [eclipse-rdf4j/rdf4j#4221] [Ideas]
Hi,

did someone ever tried to implement the SERVICE wikibase:label syntax? It is widely used in Wikidata and it might be interesting for other projects too. I refer more explicitly to:

SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en". }

Salut
D063520

**Comment by kenwenzel** [comment]
This is possible by implementing the [FederatedService](https://rdf4j.org/javadoc/4.1.0/org/eclipse/rdf4j/query/algebra/evaluation/federation/FederatedService.html) interface.

We use federated services, for example, to query time series data with SPARQL: https://github.com/linkedfactory/linkedfactory-pod/blob/c0feec58a38fa891cd385dfff6e908e21ba94627/bundles/io.github.linkedfactory.service/src/main/scala/io/github/linkedfactory/service/rdf4j/KvinFederatedService.java#L24

**Comment by D063520** [comment]
For reference, wikidata pointed me to this code which should be the actual implementation in blazegraph:

https://github.com/wikimedia/wikidata-query-rdf/tree/master/blazegraph/src/main/java/org/wikidata/query/rdf/blazegraph/label/

**Reply by JervenBolleman** [reply]
Careful, this code is likely to be GPL2 and needs an author of that code to accept integration into the RDF4j project.
